src/librustdoc/html/markdown.rs

```diff
@@ -1255,7 +1255,10 @@ crate struct MarkdownLink {
     pub range: Range<usize>,
 }
 
-crate fn markdown_links(md: &str) -> Vec<MarkdownLink> {
+/// This function returns the list of links that could be potentially intra-doc links.
+///
+/// To be noted, only `[]` links are included, `<>` links are always excluded.
+crate fn potential_intra_doc_markdown_links(md: &str) -> Vec<MarkdownLink> {
```

Comment:
```suggestion
crate fn intra_doc_markdown_links(md: &str) -> Vec<MarkdownLink> {
```
With https://github.com/rust-lang/rust/pull/96135 preprocessing is done as a part of this function and all the returned links are indeed intra doc links.

Reply:
I'll keep the `kind` pattern matching but I'll remove the other changes.